tests/components/forked_daapd/test_media_player.py

```diff
@@ -0,0 +1,726 @@
+    await updater_update(["player", "outputs", "queue"])
+    await hass.async_block_till_done()
+
+    async def add_to_queue_side_effect(
+        uris, playback=None, playback_from_position=None, clear=None
+    ):
+        await updater_update(["queue", "player"])
+
+    mock_api.return_value.add_to_queue.side_effect = (
+        add_to_queue_side_effect  # for play_media testing
+    )
+
+    async def pause_side_effect():
+        await updater_update(["player"])
+
+    mock_api.return_value.pause_playback.side_effect = pause_side_effect
+
+    return mock_api.return_value
+
+
+async def test_unload_config_entry(hass, config_entry, mock_api_object):
+    """Test the player is removed when the config entry is unloaded."""
+    assert hass.states.get(TEST_MASTER_ENTITY_NAME)
+    assert hass.states.get(TEST_ZONE_ENTITY_NAMES[0])
+    await config_entry.async_unload(hass)
+    assert not hass.states.get(TEST_MASTER_ENTITY_NAME)
+    assert not hass.states.get(TEST_ZONE_ENTITY_NAMES[0])
+
+
+def test_master_state(hass, mock_api_object):
+    """Test master state attributes."""
+    state = hass.states.get(TEST_MASTER_ENTITY_NAME)
+    assert state.state == STATE_PAUSED
+    assert state.attributes[ATTR_FRIENDLY_NAME] == "forked-daapd server"
+    assert state.attributes[ATTR_SUPPORTED_FEATURES] == SUPPORTED_FEATURES
+    assert not state.attributes[ATTR_MEDIA_VOLUME_MUTED]
+    assert state.attributes[ATTR_MEDIA_VOLUME_LEVEL] == 0.2
+    assert state.attributes[ATTR_MEDIA_CONTENT_ID] == 12322
+    assert state.attributes[ATTR_MEDIA_CONTENT_TYPE] == MEDIA_TYPE_MUSIC
+    assert state.attributes[ATTR_MEDIA_DURATION] == 0.05
+    assert state.attributes[ATTR_MEDIA_POSITION] == 0.005
+    assert state.attributes[ATTR_MEDIA_TITLE] == "Short TTS file"
+    assert state.attributes[ATTR_MEDIA_ARTIST] == "Google"
+    assert state.attributes[ATTR_MEDIA_ALBUM_NAME] == "No album"
+    assert state.attributes[ATTR_MEDIA_ALBUM_ARTIST] == "The xx"
+    assert state.attributes[ATTR_MEDIA_TRACK] == 1
+    assert not state.attributes[ATTR_MEDIA_SHUFFLE]
+
+
+async def _service_call(
+    hass, entity_name, service, additional_service_data=None, blocking=True
+):
+    if additional_service_data is None:
+        additional_service_data = {}
+    return await hass.services.async_call(
+        MP_DOMAIN,
+        service,
+        service_data={ATTR_ENTITY_ID: entity_name, **additional_service_data},
+        blocking=blocking,
+    )
+
+
+async def test_zone(hass, mock_api_object):
+    """Test zone attributes and methods."""
+    zone_entity_name = TEST_ZONE_ENTITY_NAMES[0]
+    state = hass.states.get(zone_entity_name)
+    assert state.attributes[ATTR_FRIENDLY_NAME] == "forked-daapd output (kitchen)"
+    assert state.attributes[ATTR_SUPPORTED_FEATURES] == SUPPORTED_FEATURES_ZONE
+    assert state.state == STATE_ON
+    assert state.attributes[ATTR_MEDIA_VOLUME_LEVEL] == 0.5
+    assert not state.attributes[ATTR_MEDIA_VOLUME_MUTED]
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5518609)